packages/components/app/components/hds/side-nav/button.js

```diff
@@ -0,0 +1 @@
+export { default } from '@hashicorp/design-system-components/components/hds/side-nav/button';
```

Comment:
TBD if we want to have it under `header` to avoid confusion (otherwise consumers may think it can be used in any part of the sidenav)

Reply:
Would it necessarily be bad if they used it in another part of the SideNav? What if, for example, they didn't use a `List` in the body section and instead wanted custom content including a dropdown for some reason?

Reply:
Decided makes sense to leave it as more generic element